src/main/java/gov/usgs/earthquake/distribution/DefaultNotificationSender.java

```diff
@@ -0,0 +1,215 @@
+
+
+  /**
+   * Configures based on configuration section.
+   *
+   * @param config
+   *            The config
+   * @throws Exception
+   */
+  public void configure(Config config) throws Exception {
+    // let default notification listener configure itself
+    super.configure(config);
+
+    if (getNotificationIndex() == null) {
+      throw new ConfigurationException("[" + getName()
+              + "] 'index' is a required configuration property");
+    }
+
+    String productStorageName = config.getProperty(PRODUCT_STORAGE_PROPERTY);
+    if (productStorageName == null) {
+      throw new ConfigurationException("[" + getName() + "] '" + PRODUCT_STORAGE_PROPERTY + "' is a required property.");
+    }
+    LOGGER.config("[" + getName() + "] loading product storage '" + productStorageName + "'.");
+    this.productStorage = (URLProductStorage) Config.getConfig().getObject(productStorageName);
+    if (productStorage == null) {
+      throw new ConfigurationException("[" + getName() + "] product storage '" + productStorageName + "' improperly configured.");
+    }
+
+    productStorageMaxAge = Long.parseLong(config.getProperty(PRODUCT_STORAGE_MAX_AGE_PROPERTY,DEFAULT_PRODUCT_STORAGE_MAX_AGE));
+    LOGGER.config("[" + getName() + "] product storage max age: " + productStorageMaxAge + "ms");
+
+    serverHost = config.getProperty(SERVER_HOST_PROPERTY);
+    LOGGER.config("[" + getName() + "] messenger server host: " + serverHost);
+
+    serverPort = config.getProperty(SERVER_PORT_PROPERTY);
+    LOGGER.config("[" + getName() + "] messenger server port: " + serverPort);
+  }
+
+  /**
+   * Called on receipt of a new product. Stores this product and calls sendMessage()
+   * Most of this logic was lifted from the pre-08/2019 EIDSNotificationSender class.
+   *
+   * @param product
+   *            a product whose notification was accepted.
+   * @throws Exception
+   */
+  public void onProduct(final Product product) throws Exception {
+    ProductId id = product.getId();
+
+    // store product
+    try {
+      productStorage.storeProduct(product);
+    } catch (ProductAlreadyInStorageException e) {
+      // ignore
+    }
+
+    // create notification
+    // make expiration relative to now
+    Date expirationDate = new Date(new Date().getTime()
+            + productStorageMaxAge);
+    URLNotification notification = new URLNotification(id, expirationDate,
+            product.getTrackerURL(), productStorage.getProductURL(id));
+
+    // remove any existing notifications, generally there won't be any
+    Iterator<Notification> existing = getNotificationIndex()
+            .findNotifications(id).iterator();
+    while (existing.hasNext()) {
+      getNotificationIndex().removeNotification(existing.next());
+    }
+
+    // add created notification to index. Used to track which products
+    // have been processed, and to delete after expirationDate
+    getNotificationIndex().addNotification(notification);
```

Comment:
Removes product from storage on sendNotification failure, then throws the failure exception so it can be handled upstream. Indexing is now done after sendNotification.